types/types.go

```diff
@@ -561,6 +561,8 @@ type SystemContext struct {
 	UserShortNameAliasConfPath string
 	// If set, short-name resolution in pkg/shortnames must follow the specified mode
 	ShortNameMode *ShortNameMode
+	// If set, short names will resolve docker.io, and unqualified-search registries and short-name alias are ignored.
+	ResolveShortNamesToDockerHub bool
```

Comment:
So how scary can we make this name?
`PodmanDockerEngineAPICompatOnlyResolveShortNamesToDockerHub` 😛 ? I guess not _that_, but it should stick out at least as much as …InsecureSkipTLSVerify does.

Also the documentation should
- refer to pkg/shortname
- explicitly record that this is only intended to help implementing the Docker Engine API,
- warn that this should NEVER be used by any new features and for any other purpose

Reply:
… and possibly even  warn that the exact semantics may change to support that use case only, and only as implemented by Podman + c/common. I worry a bit that e.g. the `localhost/` part is not yet 100% certain, and we might discover that we need to change the behavior only after making a c/image release.

Alternatively, name it `UnstableUnsupported…` and explicitly exclude it from the general c/image API stability effort.

Either way, if we did need to change the semantics, c/image + the callers would end up with tightly coupled versions. Or would we actually drop this field, and introduce a differently-named field for the new semantics?